bin/adhoc-scripts/fixWorkflowParentage.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+This script runs the same logic as the one executed by the StepChainParentage
+CherryPy thread, available under:
+https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/CherryPyThreads/StepChainParentageFixTask.py
+
+However, it takes a workflow name as input and performs the parentage fix only
+against that one workflow.
+Use case for this is on workflows that require a very large memory footprint.
+"""
+import time
+import logging
+import os
+import sys
+import psutil
+
+from WMCore.Services.DBS.DBS3Reader import DBS3Reader
+from WMCore.Services.RequestDB.RequestDBWriter import RequestDBWriter
+
+
+REQMGR_URL = 'https://cmsweb.cern.ch/couchdb/reqmgr_workload_cache'
+DBS_URL = 'https://cmsweb.cern.ch/dbs/prod/global/DBSWriter'
+
+
+def getChildDatasetsForStepChainMissingParent(reqmgrDB, wflowName):
+    """
+    Similar to the method implemented in the CherryPy thread.
+    :param reqmgrDB: object to the RequestDBWriter class
+    :param wflowName: string with the workflow name
+    :return: a dictionaries with request name and children datasets
+    """
+    data = reqmgrDB.getRequestByNames(wflowName)
+    for reqName, info in data.items():
+        childrenDsets = []
+        for _stepNum, dsetDict in info.get("ChainParentageMap", {}).items():
+            childrenDsets.extend(dsetDict['ChildDsets'])
+        wflowChildrenDsets = {"workflowName": reqName, "childrenDsets": childrenDsets}
+    return wflowChildrenDsets
+
+
+def updateParentageFlag(reqName, reqmgrDB, logger):
+    """
+    Given a workflow name, update its ParentageResolved flag in
+    ReqMgr2.
+    :param reqName: string with the workflow name
```

Comment:
not all input parameters are documented here, e.g. `reqmgrDB` and `logger` are missing in docstring.